rakan/cRakan/src/Runner.h

```diff
@@ -134,74 +162,100 @@ class Runner {
   * 
   * @return the score of this redistricting
   */
   double Redistrict(Node *node, int new_district);
 
-  /*
+  /**
   * Walks along the graph this Runner has loaded. Implements the
   * Metropolis-Hastings algorithm on the graph a given number of times.
   * 
-  * @param    num_steps     The number of steps to take in this walk
+  * @param    num_steps     the number of steps to take in this walk
+  * @param    alpha         the weight to assign to the compactness score
+  * @param    beta          the weight to assign to the population distribution
+  *                         score
+  * @param    gamma         the weight to assign to the existing border
+  *                         respectfulness score
+  * @param    eta           the weight to assign to the VRA score
   * 
   * @return a sum of all the scores accumulated during the walk
   */
-  double Walk(uint32_t num_steps, double alpha, double beta, double gamma, double eta);
-
+  double Walk(uint32_t num_steps,
+              double alpha, double beta, double gamma, double eta);
 
 
  //////////////////////////////////////////////////////////////////////////////
  // Queries
  //////////////////////////////////////////////////////////////////////////////
 
-  /*
+  /**
   * Queries whether or not a district is empty.
   * 
-  * @param  district   The district to query
+  * @param  district   the district to query
   * 
   * @return true iff the district is empty
   */
   bool IsEmptyDistrict(int district);
 
-  /*
+  /**
   * Queries whether or not the district that the proposed node is in will be
   * severed once the proposed node is removed.
   * 
-  * @param    proposed_node   The node that will be hypothetically removed
+  * @param    proposed_node   the node that will be hypothetically removed
   *                           from its district
   * 
   * @return true iff the district will be severed
   */
   bool IsDistrictSevered(Node *proposed_node);
 
-  /*
+  /**
   * Queries whether or not a path exists between the start node and the target
   * node. Path is only valid if all nodes traversed in the path are in the
   * same district.
   *
-  * @param   start   The node to start the search at
-  * @param   target  The node to look for
+  * @param        start        the node to start the search at
+  * @param        target       the node to look for
   *
   * @return true iff a path exists between start and target and nodes traversed
   * belong in the same district
   */
   bool DoesPathExist(Node *start, Node *target);
 
+  /**
+   * Gets the list of changes of the graph on every step of the walk. The
+   * changes are represented as a vector of uint32_t, where each index of the
+   * vector represents the precinct ID, and the value at the index represents
+   * the new district ID of that precinct.
+   * 
+   * @return a vector of vector of uint32_t, representing changes at every step
+   */
+  vector<vector<uint32_t>&>& getMaps();
```

Comment:
Same thing as `getScores()`. Are you guys responsible for cleaning up the heap?

Reply:
Yes